server/gc/src/main/java/org/apache/accumulo/gc/GCRun.java

```diff
@@ -176,9 +176,10 @@ public Stream<Reference> getReferences() {
     if (level == Ample.DataLevel.ROOT) {
       tabletStream = Stream.of(context.getAmple().readTablet(RootTable.EXTENT, DIR, FILES, SCANS));
     } else {
-      var tabletsMetadata = TabletsMetadata.builder(context).scanTable(level.metaTable())
-          .checkConsistency().fetch(DIR, FILES, SCANS).build();
-      tabletStream = tabletsMetadata.stream();
+      try (TabletsMetadata tm = TabletsMetadata.builder(context).scanTable(level.metaTable())
```

Comment:
Can not close `tm` because a stream derived from it is used to build other streams which are returned.

Reply:
Addressed in cc9bc8a by adding an `onClose()` which closes the `TabletsMetadata` and made sure the places using this method close the stream.